[FIX] server: fix tree dropping manifest dunder

mmahrouss opened the pull request.

fda-odoo reviewed (changes requested):
remove drop_dunder parameter as it should be always true. Fix where you put false

fda-odoo commented:
@mmahrouss updated, can you review it?

fda-odoo committed:
[FIX] server: custom entrypoint should not see manifest and init files

entyrpoint should be the directory if the file is init or manifest to stay coherent with odoo structure

mmahrouss reviewed (commented):
LGTM:)

fda-odoo commented:
@mmahrouss I fixed a deletion, create, rename issues. But I think we should continue to play with it to be sure it's working

mmahrouss committed:
[FIX] server: fix manifest validity check

fda-odoo committed:
[FIX] server: clean must_reload_path queue if invalid weak

fda-odoo committed:
[FIX] server: avoid directory duplication in custom entry point

fda-odoo committed:
[FIX] server: change str.starts_with with PathBuf::starts_with

This change avoid matching "dir1" as a subidrectory of "dir" as it starts with the same string. PathBuf will match on path components instead
